src/python/pants/pantsd/pants_daemon.py

```diff
@@ -164,6 +165,9 @@ def _run_services(self, services):
         self.shutdown(service_thread_map)
         raise self.StartupFailure('service {} failed to start, shutting down!'.format(service))
 
+    # Once all services are strted, write our pid.
```

Comment:
started

Also, does it make sense to wait until everything starts successfully before writing the PID? If you fail to start up, you'd still want to be killable.

Reply:
today, yes. any service failure at or just after startup will automatically cause the daemon to teardown by design. once the pid is written, the thin client will immediately try to connect - so it's important to wait until everything is successfully started otherwise we'll be racing a pailgun run vs the teardown.